### PR TITLE
Add test implementation of Google Analytics

### PR DESCRIPTION
**Pull request by gklopper**
# My Awesome Pull Request
## What does this change?

This allows us to do a limited evaluation rollout of Google Analytics. We want a chance to play with some real data and have a look at things like any effect on page performance.
## What is the value of this and can you measure success?

We are considering moving to GA. We are going to let people play with this data and see if it improves their use of analytics.

The audience team is in on this too.
## Does this affect other platforms - Amp, Apps, etc?

Not directly. Apps is aware we are doing this, and the Android app already has GA
## Screenshots

image
## 

*Does this PR meet the [contributing guidelines]

### CODE REVIEW

File: common/app/views/support/GoogleAnalyticsAccount.scala [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package views.support
+
+import conf.Configuration.environment
+import model.Page
+import conf.switches.Switches.GoogleAnalyticsSwitch
+
+object GoogleAnalyticsAccount {
+
+  private val prod = "UA-33592456-1"
```
Comment (rich-nguyen) [inline common/app/views/support/GoogleAnalyticsAccount.scala]: Are these ok public?

Comment (gklopper) [inline common/app/views/support/GoogleAnalyticsAccount.scala]: Yes, they appear in the source of the page

**Comment by OliverJAsh** [comment]
:+1: !

File: common/app/templates/inlineJS/blocking/googleAnalytics.scala.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    We are more interested in "users" than in raw page views.
+    *@
+    if (window.localStorage) {
+        var gaStorageKey = 'gu.ga.participation';
+        gaParticipation = window.localStorage.getItem(gaStorageKey) || gaParticipation;
+        window.localStorage.setItem(gaStorageKey, gaParticipation);
+    }
+} catch (ex) {
+    @* if for some reason we cannot read/write to localStorage don't be in the test *@
+    gaParticipation = 'out';
```
Comment (OliverJAsh) [inline common/app/templates/inlineJS/blocking/googleAnalytics.scala.js]: Should we log the error? `console.error(error.stack);`

**Comment by sndrs** [comment]
yay! :+1: 

@dominickendrick reminds me we should remove the footer analytics switch when this merges - this shouldn't be blocking

**Comment by sndrs** [comment]
plus we can finally enable this: http://caniuse.com/#stats_import :gem:

**Comment by OliverJAsh** [comment]
> plus we can finally enable this: http://caniuse.com/#stats_import

That would be super useful!

File: common/app/templates/inlineJS/blocking/googleAnalytics.scala.js (hunk 2 above)
Comment (gklopper) [inline common/app/templates/inlineJS/blocking/googleAnalytics.scala.js]: Mozilla warns me against using `console.error` https://developer.mozilla.org/en-US/docs/Web/API/Console/error

Comment (sndrs) [inline common/app/templates/inlineJS/blocking/googleAnalytics.scala.js]: it's not strictly an error, it's more of a safe way to feature detect – does it need an error message?

Comment (OliverJAsh) [inline common/app/templates/inlineJS/blocking/googleAnalytics.scala.js]: I was thinking more from the point of debuggability. Imagine this code isn't working but you have no error to say why.

> Mozilla warns me against using console.error

We use it along with other `console` methods frequently, but we tend to guard the statement with `if (console && console.log) …`. In practice the only JS runtime that doesn't have it is old IE.

**Pull request merged**